src/3D/Extent2D.h

```diff
@@ -0,0 +1,26 @@
+/*******************************************************************************
+ * Copyright (c) 2018-2023 openblack developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/openblack/openblack
+ *
+ * openblack is licensed under the GNU General Public License version 3.
+ *******************************************************************************/
+
+#pragma once
+
+#include <glm/mat4x4.hpp>
+
+namespace openblack
+{
+struct Extent2D
+{
+	glm::vec2 minimum;
+	glm::vec2 maximum;
+};
+struct IndexExtent
+{
+	glm::u16vec2 minimum;
+	glm::u16vec2 maximum;
+};
+} // namespace openblack
```

Comment:
Extent2D works well within `src/3D` but IndexExtent seems a little too specific to LandIsland.

What you could do is extend `glm` to have a template for all kinds of extents

```suggestion
#include <glm/detail/qualifier.hpp>

namespace openblack
{
template<glm::length_t L, typename T, glm::qualifier Q = glm::defaultp>
struct Extent
{
	glm::vec<L, T, Q> minimum;
	glm::vec<L, T, Q> maximum;
};

using Extent2 = Extent<2, float>;
using U16Extent2 = Extent<2, uint16_t>;

} // namespace openblack
```
The file would have to be renamed to Extent.